test/cpp/jit/test_lite_interpreter.cpp

```diff
@@ -703,7 +718,7 @@ TEST(LiteInterpreterTest, DuplicatedClassTypeModuleInfo) {
   )JIT");
 
   std::stringstream ss;
-  b._save_for_mobile(ss, {}, true);
+  b._save_for_mobile(ss, {}, caffe2::serialize::kProducedBytecodeVersion, true);
```

Comment:
Can we make it truly optional: i.e. no have to pass it when kProducedBytecodeVersion is used?

I think it's more of a niche use case that people will use the backport functionality, and it can be error prone if people mistakenly start hardcoding a value here.

So it's much better if basically _save_for_mobile is always used as before, except for the few cases when an actual older version is needed.

Reply:
Actually this one should be optional. The reason why we need to specify version here is that, the last argument `save_mobile_debug_info` is also optional and defaults to `false`. In this unittest, the last argument needs to be set to `true`. Since version is the second last optional argument, we need to specify the version in this case. 

If we swap the last two arguments, when exporting old model version, we would need to call it like
```
b._save_for_mobile(ss, {}, false, 4)
```
comparing to when we need `save_mobile_debug_info`, we need to call it like
```
b._save_for_mobile(ss, {}, caffe2::serialize::kProducedBytecodeVersion, true)
```

I guess ideally we would like to set variable only when needed. [Named Parameter](https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/Named_Parameter) might be a way to resolve a problem. But it may make the code more complicated. Keep the parameter order like this seems reasonable to me. What do you think?

Reply:
I add a python unittest `test_forward_compatibility.py` in the latest commit. Hopefully the api is clearer

Reply:
It is little hard to test FC because what we need is older runtime that handles new model? Latest master will always be newer than older runtime. Also the changes in this PR are not exactly FC because the serialization format change is such that older runtime cannot load the v5 model. v5 will have to converted to v4.

Reply:
Sorry it was a bit confusing. Probably `test_export_different_model_versions.py` is a better name to describe the unittest.